tests/unittests/std_interoperability.cpp

```diff
@@ -1342,7 +1342,17 @@ TEST_CASE("Collection as range", "[collection][ranges][std]") {
   // std::range::common_range
   STATIC_REQUIRE(std::ranges::common_range<CollectionType>);
   // std::range::viewable_range
+  // special handling of gcc 11
+#if defined(__GNUC__)
+  #if __GNUC__ < 12
+  auto view = std::ranges::ref_view(coll);
+  STATIC_REQUIRE(std::ranges::viewable_range<decltype(view)>);
+  #else
   STATIC_REQUIRE(std::ranges::viewable_range<CollectionType>);
+  #endif
+#else
+  STATIC_REQUIRE(std::ranges::viewable_range<CollectionType>);
+#endif
```

Comment:
```suggestion
#if defined(__GNUC__) && __GNUC__ < 12
  DOCUMENTED_STATIC_FAILURE(std::ranges::viewable_range<CollectionType>);
#else
  STATIC_REQUIRE(std::ranges::viewable_range<CollectionType>);
#endif
```


Since this is basically checking whether collections fulfill `viewable_range` concept for sake of documentation, I think I'd prefer to expect them to fail the test for GCC 11 (and add comment in docs) rather than test a workaround

Reply:
Yeah that would also be a possibility. I went with the workaround for now, because I wanted to have an gcc 11 build in CI. But another possibility would be to ensure that things build in CI and we skip the tests for gcc 11.

Reply:
The two test cases that are failing are now tagged with `[GCC11-FAIL]` and we inject a filter flag for running the tests when we build with gcc11.